charts/ocis/templates/_common/labels/labels.tpl

```diff
@@ -25,6 +25,13 @@ app.kubernetes.io/managed-by: {{ .Release.Service }}
 {{- if .Values.extraLabels }}
 {{ toYaml .Values.extraLabels }}
 {{- end }}
+
+{{- if .appSpecificConfig }}
+{{- if (index .appSpecificConfig "extraLabels") }}
```

Comment:
Cosmetic issue: I would combine them:

```suggestion
{{- if and .appSpecificConfig (index .appSpecificConfig "extraLabels" ) }}
```

Or even
```suggestion
{{- if and .appSpecificConfig .appSpecificConfig.extraLabels }}
```
Or use `with` to avoid repetition:

```
{{- with and .appSpecificConfig .appSpecificConfig.extraLabels }}
{{ toYaml . }}
{{- end }}
```

I would prefer the last one.